hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/zeroconfig/ValueReaderWriters.java

```diff
@@ -211,7 +211,8 @@ public static ValueReaderWriter<?> readerWriterFor(CompactStreamSerializer compa
         }
 
         // The nested field might not be Compact serializable
-        verifyFieldClassIsCompactSerializable(type, clazz);
+        boolean isRegisteredAsCompact = compactStreamSerializer.isRegisteredAsCompact(type);
+        verifyFieldClassIsCompactSerializable(type, clazz, isRegisteredAsCompact);
         verifyFieldClassShouldBeSerializedAsCompact(compactStreamSerializer, type, clazz);
```

Comment:
Why do we perform these checks if the class is registered as Compact serializable?

Reply:
If think we can keep the check `verifyFieldClassIsCompactSerializable`, because if we remove it as well, the check for `Void`, array of array and primitive in `canBeSerializedAsCompact` would be skipped. Do we want to serialize Void or array of array if there's a explicit serializer for them? 

we can skip the second check IMO which is `verifyFieldClassShouldBeSerializedAsCompact`. WDYT?

Reply:
They can't add serializers for primitive types. And, for the other two, if they really wanted that and added serializers, I believe it is OK from our end to respect that

Reply:
Ok, I applied your suggestion and pushed.

Reply:
Still, I wonder what is @ihsandemir 's  view on this.

Reply:
@ihsandemir any comment on this?

Reply:
my view: If the user put the compact serializer for any type, let us just trust the user and use it.